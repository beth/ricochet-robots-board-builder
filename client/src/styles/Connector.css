:local .set-connector {
  height: 10px;
  width: 10px;
  margin: auto;
}

:local .red {
  background-color: red;
}

:local .blue {
  background-color: blue;
}

:local .yellow {
  background-color: yellow;
}

:local .green {
  background-color: green;
}