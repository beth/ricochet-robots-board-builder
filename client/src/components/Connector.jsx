import React from 'react';
import PropTypes from 'prop-types';
import COLORS from '../lib/config/colors';
import ColorsContainer from '../containers/ColorsContainer';
import styles from '../styles/Connector.css';

const connectorStyles = color => `${styles['set-connector']} ${styles[color.toLowerCase()]}`;

const Connector = ({ color, row, col }) => (
  COLORS.includes(color) ?
    <div className={connectorStyles(color)} /> :
    <ColorsContainer row={row} col={col} />
);

Connector.propTypes = {
  row: PropTypes.number.isRequired,
  col: PropTypes.number.isRequired,
  color: PropTypes.string.isRequired,
};

export default Connector;
