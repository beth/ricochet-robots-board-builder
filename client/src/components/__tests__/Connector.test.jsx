import React from 'react';
import renderer from 'react-test-renderer';
import Connector from '../Connector';

jest.mock('../../containers/ColorsContainer', () => 'ColorsContainer');

describe('Connector Component', () => {
  it('renders a plain div when given a valid color', () => {
    const domTree = renderer.create(<Connector
      color="RED"
      row={8}
      col={8}
    />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
  it('renders a ColorsContainers component when given an empty string', () => {
    const domTree = renderer.create(<Connector
      color=""
      row={8}
      col={8}
    />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
