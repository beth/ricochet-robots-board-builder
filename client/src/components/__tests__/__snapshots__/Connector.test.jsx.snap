// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connector Component renders a ColorsContainers component when given an empty string 1`] = `
<ColorsContainer
  col={8}
  row={8}
/>
`;

exports[`Connector Component renders a plain div when given a valid color 1`] = `
<div
  className="set-connector red"
/>
`;
